lib/rules/object-shorthand.js

```diff
@@ -497,6 +497,13 @@ module.exports = {
                         node,
                         messageId: "expectedPropertyShorthand",
                         fix(fixer) {
+                            const lastKeyToken = sourceCode.getLastToken(node.key);
+
+                            // x: /* */ x
+                            if (sourceCode.commentsExistBetween(lastKeyToken, node.value)) {
+                                return null;
+                            }
+
```

Comment:
how about if we give a fix here something like:
```js
var x = {
a /* comment */: a,
b
}
```
can be fixed to
```js
var x = {
/* comment */ a,
b
}
```
or

```js
var x = {
a, /* comment */ 
b
}
```

Reply:
I think in that case it becomes a feature request, not a bug fix. You can create an issue for this idea.

Whatever the decision is, I'd recommend that both identifiers and function expressions were handled identically (which they aren't right now, hence this PR).

Reply:
I think it's fine not to provide the fix when there's a comment between the two identifiers and let the author fix this manually as it isn't clear where the new position for the comment should be.